File: Euclid/Messages/Outgoing/FLATPROPERTY.cs
using Euclid.Network.Streams.Util;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Euclid.Messages.Outgoing
{
    class FLATPROPERTY : IMessageComposer
    {
        private string sprite;
        private string customData;

        public FLATPROPERTY(string sprite, string customData)
        {
            this.sprite = sprite;
            this.customData = customData;
        }

        public override void Write()
        {
            this.Data.Add(new ArgumentEntry(this.sprite));
            this.Data.Add(new DelimeterEntry(this.customData, "/"));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/FLATPROPERTY.java
package euclid.messages.outgoing;

import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;
import euclid.network.streams.util.DelimeterEntry;

public class FLATPROPERTY extends IMessageComposer {

    private final String sprite;
    private final String customData;

    public FLATPROPERTY(String sprite, String customData) {
        this.sprite = sprite;
        this.customData = customData;
    }

    @Override
    public void write() {
        getData().add(new ArgumentEntry(sprite));
        getData().add(new DelimeterEntry(customData, "/"));
    }
}
